### PR TITLE
fix: fix upload repetitive key into the Rafiki

### LINKED ISSUE #1460: Developer Keys duplicate keys uploade bug
## Bug Report

**Describe the bug**

Uploading the same key multiple times gives you correct error message, that you need to upload a unique one, but the backend API for uploading the key is still being called.

**To Reproduce**

- Upload a key in Developer Keys on Test Wallet, and check the API call.
- Upload the same key with a different nickname, and you will see the error message to upload a unique key, but check the API call again. The key is being uploaded.

**Expected behavior**

When a duplicate key is being uploaded backend API call for upload should not be called.

### PR DESCRIPTION
**Pull request by rico191013**
### Context

- fixes #1460

### CODE REVIEW

File: packages/wallet/backend/src/walletAddressKeys/service.ts [hunk 1]
```diff
@@ -78,6 +77,15 @@ export class WalletAddressKeyService implements IWalletAddressKeyService {
       .export({ type: 'spki', format: 'pem' })
       .toString()
 
+    const isExist = await WalletAddress.query().where({
```
Comment (Tymmmy) [inline packages/wallet/backend/src/walletAddressKeys/service.ts:80]:
```suggestion
    const isUploaded = await WalletAddress.query().where({
```

File: packages/wallet/backend/src/walletAddressKeys/service.ts [hunk 2]
```diff
@@ -78,6 +77,15 @@ export class WalletAddressKeyService implements IWalletAddressKeyService {
...
+      id: jwk.kid.toString()
+    })
+
+    if (isExist)
```
Comment (Tymmmy) [inline packages/wallet/backend/src/walletAddressKeys/service.ts:84]:
```suggestion
    if (isUploaded)
```

File: packages/wallet/frontend/src/pages/grant-interactions/index.tsx [hunk 3]
```diff
@@ -175,7 +175,7 @@ export const getServerSideProps: GetServerSideProps<{
     if (access.limits) {
       access.limits.receiver = access.limits.receiver
         ? replaceWalletAddressProtocol(access.limits.receiver)
-        : (access.limits.receiver ?? null)
+        : access.limits.receiver ?? null
```
Comment (Tymmmy) [inline packages/wallet/frontend/src/pages/grant-interactions/index.tsx:178]: I think new prettier needs the ()

**Review by Tymmmy — APPROVED** [review]

**Review by dragosp1011 — APPROVED** [review]

**Review by dragosp1011 — CHANGES_REQUESTED** [review]
please fix failed tests

**Review by dragosp1011 — APPROVED** [review]

**Review by Tymmmy — APPROVED** [review]

**Pull request merged**